data/npc/lib/npcsystem/modules.lua

```diff
@@ -700,7 +700,7 @@ if Modules == nil then
 				i = i + 1
 			end
 
-			if name ~= nil and container ~= nil and itemid ~= nil and cost ~= nil then
+			if name and container and itemid and cost then
```

Comment:
Not related to your changes, but I think "container" is never set. It might be a bug in the npc system (itemid is set in 2 cases).